### PR TITLE
Add pointer to CriticalUp docs.

### PR DESCRIPTION
**Pull request by amanjeev**
This is a quick PR to add link to CriticalUp instead of placeholder text.

Ticket: https://ferroussystems.clickup.com/t/86947z6md

### CODE REVIEW

File: ferrocene/doc/user-manual/src/rustc/install.rst [hunk 1]
```diff
@@ -21,11 +21,15 @@ section of the target documentation.
...
+``rustup`` like tool called ``criticalup``.
+
+Find more details about using ``criticalup`` in the
+`CriticalUp User Documentation <https://criticalup.ferrocene.dev/index.html>`_.
 
-   We are developing a tool similar to ``rustup``, able to manage your
-   Ferrocene installations and install Ferrocene without relying on a web
-   browser. It will be available soon.
+.. note::
+    Are you trying to target a specific architecture using cross-compilation?
```
Comment (Hoverbear) [inline ferrocene/doc/user-manual/src/rustc/install.rst:31]: The UM is not currently interrogative. Perhaps consider an instructive tone.

In the end, this feels redundant as immediately above it we instruct them of basically the same thing: https://docs-dev.ferrocene.dev/nightly/user-manual/rustc/install.html#prerequisites

**Review by Hoverbear — APPROVED** [review]
bors merge

**Comment by amanjeev** [comment]
bors merge

(always check branch names before mucking up your own PR, Aman)

**Review by Hoverbear — APPROVED** [review]
bors merge

**Pull request merged**